qiita_db/search.py

```diff
@@ -396,16 +401,20 @@ def filter_by_processed_data(self, datatypes=None):
                 study = qdb.study.Study(study_id)
                 study_sample_ids = {s[0] for s in study_meta}
                 study_proc_ids[study_id] = defaultdict(list)
-                for proc_data_id in study.processed_data():
-                    proc_data = qdb.data.ProcessedData(proc_data_id)
-                    datatype = proc_data.data_type()
+                for artifact in study.artifacts():
+                    if artifact.artifact_type != 'BIOM':
+                        continue
+                    datatype = artifact.data_type
                     # skip processed data if it doesn't fit the given datatypes
                     if datatypes is not None and datatype not in datatypes:
                         continue
-                    filter_samps = proc_data.samples.intersection(
```

Comment:
Yup in order to avoid this issue:
```python
In [1]: a = None

In [2]: '1' in a
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-2-096bed3860b1> in <module>()
----> 1 '1' in a

TypeError: argument of type 'NoneType' is not iterable
```